### PR TITLE
[SDPAP-6684] Verify items in search index drush command

### PR DESCRIPTION
**Pull request by MdNadimHossain**
### JIRA
https://digital-vic.atlassian.net/browse/SDPAP-6684

### Changes
1. Made a new drush command to compare the published node ids from the CMS with the search index

### Testing PR 
[https://nginx-php.pr-142.content-reference-sdp-vic-gov-au.sdp4.sdp.vic.gov.au](https://nginx-php.pr-143.content-reference-sdp-vic-gov-au.sdp4.sdp.vic.gov.au/)

### Test steps 
1. are added in the ticket
2. Also added dedicated search index for the PR so it does not get overwritten with other PR

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,12 +6,19 @@
     "require": {
         "drupal/search_api": "^1.11",
         "drupal/elasticsearch_connector": "^7.0",
-        "dpc-sdp/tide_core": "^3.0.0"
+        "dpc-sdp/tide_core": "dev-feature/SRM-512_update_drupal_to_9.4"
```
Comment (MdNadimHossain) [inline composer.json:9]: will be reverted before merging

File: dev-tools.sh [hunk 2]
```diff
@@ -33,6 +33,6 @@
 # stable version.
 #
 # Uncomment and set the Dev-Tools's commit value and commit this change.
-# export GH_COMMIT=COMMIT_SHA
+export GH_COMMIT=960eba39ed9543b17654dbd5e06f3a2ab2b1fb4b
```
Comment (MdNadimHossain) [inline dev-tools.sh:36]: will be reverted before merging

File: src/Commands/TideSearchCommands.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          $query = $indexes[$indexId]->query();
+          $query->range(0, 10000);
+          $query->sort('nid');
+          $query->addCondition('nid', $nid_starting_point, '>');
+          $results = $query->execute();
+        }
+        catch (SearchApiException $e) {
+          throw new ConsoleException($e->getMessage(), 0, $e);
+        }
+        foreach ($results->getResultItems() as $item) {
```
Comment (vincent-gao) [inline src/Commands/TideSearchCommands.php:109]: hey @MdNadimHossain ,
I think you don't need the function `convertToNodeIds`.
the `$results` is an object that implements `ResultInterface`, so I think you can get the actual result by using `$results->getExtraData('elasticsearch_response')' or `$results->getAllExtraData()`, then loop through to get `nid` from the result.

File: src/Commands/TideSearchCommands.php [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      $es_nids[] = self::convertToNodeIds($item);
+    }
+    $es_nids = array_unique($es_nids);
+    return $es_nids;
+  }
+
+  /**
+   * Helper to convert node ids.
+   */
+  public static function convertToNodeIds($item) {
```
Comment (vincent-gao) [inline src/Commands/TideSearchCommands.php:138]: hey @MdNadimHossain ,
I think you don't need the function `convertToNodeIds`.
the `$results` is an object that implements `ResultInterface`, so I think you can get the actual result by using `$results->getExtraData('elasticsearch_response')` or `$results->getAllExtraData()` , then loop through to get `nid` from the result.

**Review by vincent-gao — APPROVED** [review]

**Pull request merged**